DFC.Api.JobProfiles.IntegrationTests/Support/Common/CommonAction.cs

```diff
@@ -0,0 +1,129 @@
+            Settings.APIConfig.EndpointBaseUrl.ProfileDetail = configuration.GetSection("APIConfig").GetSection("EndpointBaseUrl").GetSection("ProfileDetail").Value;
+            Settings.APIConfig.EndpointBaseUrl.ProfileSearch = configuration.GetSection("APIConfig").GetSection("EndpointBaseUrl").GetSection("ProfileSearch").Value;
+            Settings.APIConfig.EndpointBaseUrl.ProfileSummary = configuration.GetSection("APIConfig").GetSection("EndpointBaseUrl").GetSection("ProfileSummary").Value;
+            if (!int.TryParse(configuration.GetSection("GracePeriodInSeconds").Value, out int gracePeriodInSeconds))
+            {
+                throw new InvalidCastException("Unable to retrieve an integer value for the grace period setting");
+            }
+
+            Settings.GracePeriod = TimeSpan.FromSeconds(gracePeriodInSeconds);
+            if (!int.TryParse(configuration.GetSection("DeploymentWaitInMinutes").Value, out int deploymentWaitInMinutes))
+            {
+                throw new InvalidCastException("Unable to retrieve an integer value for the deployment wait setting");
+            }
+
+            Settings.DeploymentWaitInMinutes = TimeSpan.FromMinutes(deploymentWaitInMinutes);
+        }
+
+        public async Task DeleteJobProfileWithId(Topic topic, Guid jobProfileId)
+        {
+            JobProfileContentType messageBody = ResourceManager.GetResource<JobProfileContentType>("JobProfileDelete");
+            messageBody.JobProfileId = jobProfileId.ToString();
+            Message deleteMessage = this.CreateDeleteMessage(jobProfileId, this.ConvertObjectToByteArray(messageBody));
+            await topic.SendAsync(deleteMessage).ConfigureAwait(true);
+        }
+
+        public async Task CreateJobProfile(Topic topic, Guid messageId, string canonicalName)
+        {
+            JobProfileContentType messageBody = ResourceManager.GetResource<JobProfileContentType>("JobProfileCreate");
+            messageBody.JobProfileId = messageId.ToString();
+            messageBody.UrlName = canonicalName;
+            messageBody.CanonicalName = canonicalName;
+            Message message = this.CreateCreateMessage(messageId, this.ConvertObjectToByteArray(messageBody));
+            await topic.SendAsync(message).ConfigureAwait(true);
+        }
+
+        public async Task<Response<T>> ExecuteGetRequest<T>(string endpoint, bool authoriseRequest = true)
+        {
+            GetRequest getRequest = new GetRequest(endpoint);
+            getRequest.AddVersionHeader(Settings.APIConfig.Version);
+
+            if (authoriseRequest)
+            {
+                getRequest.AddApimKeyHeader(Settings.APIConfig.ApimSubscriptionKey);
+            }
+            else
+            {
+                getRequest.AddApimKeyHeader(this.RandomString(20).ToLower(CultureInfo.CurrentCulture));
+            }
+
+            await Task.Delay(5000).ConfigureAwait(true);
```

Comment:
What does this do and how did we diced on the 5000, is it going to create intermittent issues?

Reply:
We need a delay here so that we give each of the segments enough time to update after throwing out a job profile. I initially wrote something to query the cosmos db to check if the JP had been processed but it was decided by Muthu and Lionel that an explicit wait is sufficient. 5 seconds was decided on by running the framework numerous times and identifying roughly a 2/3 second delay.